File: gcd.c
#include "stdio.h"
#include "gcd.h"

int gcd(int a, int b)
{
	if (a == 0)
		return b;

	if (b == 0)
		return a;

	if (a < b)
		return gcd(a, (b % a));
	else if (b < a)
		return gcd(b, (a % b));
}

int gcd_of_array(int *a, int size)
{
	if (a == NULL || size == 0)
		return 0;

	int g = a[0];
	for (int i = 1; i < size; i++)
	{
		g = gcd(a[i], g);
	}

	return g;
}

File: GCD.java
public class GCD {
    int gcd(int a, int b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public static void main(String[] args) {
        GCD gcd = new GCD();

        System.out.println(gcd.gcd(25, 5));
        System.out.println(gcd.gcd(25, 6));
        System.out.println(gcd.gcd(12, 9));
        System.out.println(gcd.gcd(256, 52));
    }
}
